### ISSUE TITLE
Cannot access pods without sidecar belong to a headless service even if auto mtls is enabled

### ISSUE THREAD

**Issue by knight42** [istio/istio#23673]
**Bug description**
We configure istio to use permissive tls mode, and I notice that a pod with envoy sidecar cannot access pods without sidecar belongs to a headless service even if auto mtls is enabled.

In addition, I found the following error in envoy access log:
```
...
"upstream_transport_failure_reason":"TLS error: 268435703:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER"
...
```

Looks like that envoy sidecar still sends tls traffic to the target pods without sidecar which I think is not aligned with what auto mtls should do as described in https://istio.io/pt-br/docs/tasks/security/authentication/auto-mtls/:
> Istio tracks the server workloads migrated to Istio sidecar, and configures client sidecar to send mutual TLS traffic to those workloads automatically, and send plain text traffic to workloads without sidecars. 

[x] Networking
[x] User Experience

**Expected behavior**
Pods without sidecar should be accessible if auto mtls is enabled.

**Steps to reproduce the bug**
Create a headless service without sidecar injection, then access this headless service from another pods with sidecar.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
$ istioctl version
client version: 1.5.0
control plane version: 1.5.2
data plane version: 1.5.1 (4 proxies), 1.5.0 (2 proxies), 1.5.2 (9 proxies)

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T23:35:15Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7", GitCommit:"be3d344ed06bff7a4fc60656200a93c74f31f9a4", GitTreeState:"clean", BuildDate:"2020-02-11T19:24:46Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
`istioctl manifest apply --set profile=default`

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by knight42** [comment]
/cc @hzxuzhonghu 

May be related to #20231

**Comment by hzxuzhonghu** [comment]
FYI https://github.com/istio/istio/pull/21987

You can disable tls with a DestinationRule as a workaround

**Comment by aletundo** [comment]
@hzxuzhonghu I'm trying the same with a MongoDB ReplicaSet with headless service but it doesn't seem to work.

I specified a DestinationRule for each of the Pods of the StatefulSet but I still got

`Error receiving request from client: SSLHandshakeFailed: SSLHandshakeFailed. Ending connection from 10.16.3.149:33538 (connection id: 195431)`

in the MongoDB container log when a workload with the sidecar tries to connect to MongoDB Pods (without the sidecar).

**Comment by hzxuzhonghu** [comment]
Did you disable the tls with a DestinationRule?

**Comment by knight42** [comment]
this issue should be resolved by #21987.
closing
